sdp/processors/langs/armenian.py

```diff
@@ -0,0 +1,75 @@
+import os
+import pandas as pd
+from pathlib import Path
+from sdp.processors.base_processor import BaseProcessor, BaseParallelProcessor, DataEntry
+from sdp.utils.common import load_manifest
+
+
+class GetSourceBookName(BaseParallelProcessor):
+    """
+    Processor for extracting source book name from file paths and updating the manifest.
+
+    Args:
+        source_filepath (str): The field containing the file path in the manifest.
+        source_field (str): The field to store the extracted source book name in the manifest.
+        **kwargs: Additional keyword arguments to be passed to the base class `BaseParallelProcessor`.
+
+    """
+    def __init__(
+        self,
+        source_filepath: str,
+        source_field: str,
+        **kwargs,
+    ):
+        super().__init__(**kwargs)
+        self.source_filepath = source_filepath
+        self.source_field = source_field
+
+    def process_dataset_entry(self, data_entry):
+        input_values = os.path.splitext(data_entry[self.source_filepath])[0].split("/")
+        
+        data_entry[self.source_field] = input_values[-1]
+        return [DataEntry(data=data_entry)]
+
+
+class MakeTsv(BaseProcessor):
```

Comment:
Can you please explain why we need this and the following processor? Just trying to understand what's the use-case that requires tsv instead of jsonl?

Reply:
Common Voice requires to submit the text data in a certain tsv format